features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Listing/components/ImageGenerator/ApiThumb.jsx

```diff
@@ -122,9 +122,16 @@ class APIThumb extends Component {
      * @memberof APIThumb
      */
     render() {
-        const { classes, api, isAPIProduct } = this.props;
+        const { classes, api } = this.props;
+        let { isAPIProduct } = this.props;
         const { isHover, loading } = this.state;
-        api.apiType = API.CONSTS.API;
+        if (api.type === 'APIPRODUCT') {
+            isAPIProduct = true;
```

Comment:
We pass the isAPIProduct prop from the TableView component to identify whether its an api product or not. If we do the samething here, then there is no point of passing that prop to APIThumb child component. 
Better to set the value at the TableView component.